Sources/DbPackages-postgresql.php

```diff
@@ -473,7 +473,14 @@ function smf_db_change_column($table_name, $old_column, $column_info)
 	// Different default?
 	if (isset($column_info['default']) && $column_info['default'] != $old_info['default'])
 	{
-		$action = $column_info['default'] !== null ? 'SET DEFAULT \'' . $smcFunc['db_escape_string']($column_info['default']) . '\'' : 'DROP DEFAULT';
+		// Fix the default.
+		$default = '';
+		if (isset($column_info['default']) && is_null($column_info['default']) && empty($column_info['not_null']))
+			$default = 'NULL';
+		else
+			$default = '\'' . $smcFunc['db_escape_string']($column_info['default']) . '\'';
+
+		$action = $column_info['default'] !== null ? 'SET DEFAULT ' . $default : 'DROP DEFAULT';
```

Comment:
Maybe i miss something, but when in line 478 is it essential that $column_info['default'] is null,
to fill $default with null
and here you check if $column_info['default'] !== null  -> when null than "drop default" so it never read the $default.

Reply:
Your right.  I compared it as a string instead.